core/src/main/java/org/infinispan/manager/impl/ReplicableManagerFunctionCommand.java

```diff
@@ -40,7 +41,7 @@ public ReplicableManagerFunctionCommand(Function<? super EmbeddedCacheManager, ?
    }
 
    @Override
-   public CompletableFuture<Object> invokeAsync() throws Throwable {
+   public CompletableFuture<Object> invokeAsync(GlobalComponentRegistry globalComponentRegistry) throws Throwable {
       if (subject == null) {
          return CompletableFuture.completedFuture(function.apply(new UnwrappingEmbeddedCacheManager(manager)));
```

Comment:
you should execute the `function` in the `BlockingManager`

Reply:
Not sure why, I thought the `canBlock` below was returning `true` :laughing:.